Dev give snake brain

JNotelddim opened the pull request.

JNotelddim committed:
Use brain in api

*overloaded Coordinate constructor
*renamed data['coords'] to data['body'] to reflect actual data
*updated tests

mattcasiro commented on app/main.py:
```diff
@@ -18,23 +21,22 @@ def ping():
-    return api.start_response("#00ff00")
+    data: Any = request.json
+    print("Starting game %s" % data["game"]["id"])
+    return api.start_response("ffb6c1")
```
Color probably needs a `#`

- [x] Prefix `#`

mattcasiro commented on src/coordinate.py:
```diff
@@ -1,15 +1,43 @@
     """Abstract each board position."""
 
-    def __init__(self, coordinate: Tuple[int, int]) -> None:
+    def __init__(self, coordinate: Any, secondCoord: Optional[int]) -> None:
```
If we default the second parameter to `None` it will make it much cleaner to call `Coordinate()`.

- [x] Set `secondCoord` to default to `None`
- [x] Remove `None` from `Coordinate()` calls

mattcasiro commented on src/coordinate.py:
```diff
@@ -1,15 +1,43 @@
+
+    def fromDict(self, coordinate: dict) -> None:
+        """Instantiate Coordinate from object"""
+        if coordinate["x"] is not None and coordinate["y"] is not None:
```
I think this check can be more thorough, have you tested what happens when 'x' or 'y' don't exist? Does it fail gracefully? What if we pass in `string` instead of `int`? Or floats?

Orrr maybe we don't need to worry about bad data here, and can trust our inputs. Let me know what you think.

mattcasiro commented on src/coordinate.py:
```diff
@@ -1,15 +1,43 @@
+        #self._coordinate: Tuple[int, int] = coordinate
+        if isinstance(coordinate, tuple): 
+            self.fromTuple(coordinate)
+        elif isinstance(coordinate, int) and secondCoord is not None:
```
Same idea as the comment below - are we just going to trust our data here? If so, why check if it's `None` at all?

JNotelddim committed:
Resolve PR change requests

*add hastag to color
*add default value to second param in coordinate constructor, and
refactor all coordinate instantiations with `, None` in them
*remove questionable dict checking in Coordinate fromDict

mattcasiro reviewed (changes requested):
Let's try rebasing this off develop, to resolve the duplicate commits caused from me rudely rebasing it earlier.

JNotelddim committed:
Improve Coordinate constructor logic
